### DISCUSSION TITLE
Is docker-gen a requirement for acme-companion?

### DISCUSSION THREAD

**Discussion by edemirkan** [nginx-proxy/acme-companion#852] [Q&A] [answered]
Checking the Dockerfile here;
https://github.com/nginx-proxy/acme-companion/blob/main/Dockerfile

Why does `acme-companion` requires `docker-gen` to be build and injected into its image? If I'm to use a 3 container approach; separate containers for `nginx-proxy`, `docker-gen`, `acme-companion` can I remove the first part in that Dockerfile and just go with

```
FROM alpine:3.13.5

LABEL maintainer="Nicolas Duchon <nicolas.duchon@gmail.com> (@buchdag)"

ARG GIT_DESCRIBE
ARG ACMESH_VERSION=2.8.8
ARG ACME_COMPANION_VERSION=2.1.0

ENV COMPANION_VERSION=$GIT_DESCRIBE \
    DOCKER_HOST=unix:///var/run/docker.sock \
    PATH=$PATH:/app

# Install packages required by the image
RUN apk add --no-cache --virtual .bin-deps \
        bash \
        coreutils \
        curl \
        jq \
        openssl \
        socat

# Install acme.sh and the letsencrypt service
RUN mkdir /companion-src \
    && curl -sSL https://github.com/nginx-proxy/acme-companion/archive/refs/tags/v${ACME_COMPANION_VERSION}.tar.gz \
    | tar -C /companion-src -xz \
    && mv /companion-src/acme-companion-${ACME_COMPANION_VERSION}/app /app \
    && /companion-src/acme-companion-${ACME_COMPANION_VERSION}/install_acme.sh \
    && rm -rf /companion-src

WORKDIR /app

ENTRYPOINT [ "/bin/bash", "/app/entrypoint.sh" ]
CMD [ "/bin/bash", "/app/start.sh" ]
```

**Comment by buchdag** [comment, marked as answer]
Nope you can't, the acme-companion container relies on its own internal docker-gen process for config file rendering and process signaling.

Theoretically it should be possible to run a single docker-gen container that render the configuration file for and signal both nginx and acme-companion, but the acme-companion container was never built to work that way and would require significant changes to work like this.

That's an idea that I had for some time but never developed because that would be completely backward incompatible with existing setups.
